daffodil-runtime1/src/main/scala/org/apache/daffodil/calendar/DFDLCalendar.scala

```diff
@@ -121,7 +121,8 @@ trait OrderedCalendar { self: DFDLCalendar =>
    * can be determined.
    */
   private def orderIgnoreTimeZone(p: DFDLDateTime, q: DFDLDateTime): DFDLCalendarOrder = {
-    Assert.invariant((p.hasTimeZone && q.hasTimeZone) || (!p.hasTimeZone && !q.hasTimeZone))
+    Assert.invariant(!p.hasTimeZone || p.calendar.getTimeZone == TimeZone.GMT_ZONE)
```

Comment:
The scaladoc here is not consistent with what this method does anymore. It describes a contract that P and Q must match as to whether they have, or do not have timezone. 

In what sense does this "ignore timezone" anyway? Seems to me it treats TZ like any other field, with the exception of the two assert.invariant calls.  Is this method's name actually indicative of it's behavior?

Reply:
When doing the comparison of two dateTimes, the only fields that are inspected are year, month, day, hour, minute, and second. So in that sense the timezone is ignored. Technically, it is not completely ignored though, since the dateTime is normalized to UTC, so a dateTime with a non-UTC TZ will have its hours/minutes modified, which are compared.

So this function name ism't completely innacurate. That said, something like orderCompareFields might be more specific about what it's actually doing, since all it's really doing is looping over a subset of the fields of a calendar and comparing them. I'll update the name and comment to be accurate according to what the method does.